### PR TITLE
File connector

### PR DESCRIPTION
**Pull request by Weves**

### CODE REVIEW

File: backend/danswer/background/file_deletion.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from danswer.background.utils import interval_run_job
+from danswer.connectors.file.utils import clean_temp_files
+
+
+if __name__ == "__main__":
```
Comment (yuhongsun96) [inline backend/danswer/background/file_deletion.py:5]: Can we add a comment on the logic here, like how the logic prevents deleting files that haven't yet been indexed etc

Comment (yuhongsun96) [reply backend/danswer/background/file_deletion.py:5]: or alternatively just call it clean_processed_files or something

File: backend/danswer/background/run_all.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (yuhongsun96) [inline backend/danswer/background/run_all.sh:1]: what is this for?

File: backend/danswer/connectors/file/connector.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+from danswer.connectors.models import Section
+from danswer.utils.logging import setup_logger
+
+
+logger = setup_logger()
+
+_LINK_METADATA_FLAG = "#LINK="
+
+
+def _get_files_from_zip(
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/connector.py:26]: can we handle multiple common compression types?

File: backend/danswer/connectors/file/connector.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.utils.logging import setup_logger
+
+
+logger = setup_logger()
+
+_LINK_METADATA_FLAG = "#LINK="
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/connector.py]: I prefer explicitly mentioning danswer in it so there's even less chance of accidental hit or whatever. 
#DANSWER_METADATA={"link":"whatever"}

how do we feel about this style?

File: backend/danswer/connectors/file/connector.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            with zip_file.open(file_name, "r") as file:
+                yield os.path.basename(file_name), file
+
+
+def _open_files_at_location(
+    file_path: str | Path,
+) -> Generator[tuple[str, IO[Any]], Any, None]:
+    extension = get_file_ext(file_path)
+
+    if extension == ".zip":
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/connector.py:40]: tar.gz, .tar.bz2, rar, maybe 7zip? are common as well. Does it take much effort to support those, like does the zipfile library handle these?

File: backend/requirements/default.txt [hunk 6]
```diff
@@ -22,6 +22,7 @@ pydantic==1.10.7
 PyGithub==1.58.2
 PyPDF2==3.0.1
 pytest-playwright==0.3.2
+python-multipart==0.0.6
```
Comment (yuhongsun96) [inline backend/requirements/default.txt:25]: Is the zipfile library needed?

Comment (yuhongsun96) [reply backend/requirements/default.txt:25]: seems built in based on the import, nvm

File: backend/danswer/connectors/file/connector.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+                if len(documents) >= self.batch_size:
+                    yield documents
+                    documents = []
+
+        if documents:
+            yield documents
+
+
+if __name__ == "__main__":
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/connector.py]: not a big fan of this disgusting stuff, can we remove it?

File: backend/danswer/connectors/file/connector.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        self.file_locations = [Path(file_location) for file_location in file_locations]
+        self.batch_size = batch_size
+
+    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
+        pass
+
+    def load_from_state(self) -> GenerateDocumentsOutput:
+        documents: list[Document] = []
+        for file_location in self.file_locations:
+            files = _open_files_at_location(file_location)
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/connector.py:93]: what's the flow, how is the file location set? I assume it's some directory under the temp location? Is it going to be by the name of the zip? Plz note the flow somewhere on the backend for the frontend illiterates

File: backend/danswer/connectors/file/utils.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import os
+import shutil
+import time
+import uuid
+from pathlib import Path
+from typing import Any
+from typing import IO
+
+_TEMP_LOCATION = Path("/tmp/file_connector")
+_FILE_AGE_CLEANUP_THRESHOLD = 3  # hours
```
Comment (yuhongsun96) [inline backend/danswer/connectors/file/utils.py]: probably it's fine to not have any additional logic and if we cant index the whole zip in 3 hours its just not valid?

I'm not sure about that assumption actually...
Can we delete files as they get handled and not delete the ones that failed so we can go back and see why?

Comment (yuhongsun96) [reply backend/danswer/connectors/file/utils.py]: Ok let's not increase complexity, let's just set the cleanup age to a week

File: backend/danswer/server/manage.py [hunk 10]
```diff
@@ -7,6 +7,7 @@
 from danswer.configs.app_configs import MASK_CREDENTIAL_PREFIX
```
Comment (yuhongsun96) [inline backend/danswer/server/manage.py:7]: maybe a bit late to be pointing this out but I think django has a manage.py so it's maybe a bit weird to be calling it this... not big deal though

File: backend/supervisord.conf [hunk 11]
```diff
@@ -0,0 +1,18 @@
+[supervisord]
+nodaemon=true
+logfile=/dev/null
+logfile_maxbytes=0
+
+[program:indexing]
+command=python danswer/background/update.py
+stdout_logfile=/var/log/supervisor/update.log
+stdout_logfile_maxbytes=0
```
Comment (yuhongsun96) [inline backend/supervisord.conf]: this one spits out looottta junk, should probably roll it.

The file deletion one as long as we're not logging too much junk, I think we can keep around all the files that are uploaded/extracted/deleted

**Review by yuhongsun96 — APPROVED** [review]
Approved but please address the commented stuff

File: backend/danswer/connectors/file/connector.py (hunk 5 above)
Comment (Weves) [reply backend/danswer/connectors/file/connector.py:40]: I think it's fine for now, can add those as follow ups if needed

File: backend/danswer/background/file_deletion.py (hunk 1 above)
Comment (Weves) [reply backend/danswer/background/file_deletion.py:5]: Sure, added a more descriptive name

File: backend/danswer/background/run_all.sh (hunk 2 above)
Comment (Weves) [reply backend/danswer/background/run_all.sh:1]: https://medium.com/@codingmaths/bin-bash-what-exactly-is-this-95fc8db817bf

File: backend/danswer/connectors/file/connector.py (hunk 4 above)
Comment (Weves) [reply backend/danswer/connectors/file/connector.py]: sure, seems good

File: backend/danswer/connectors/file/connector.py (hunk 8 above)
Comment (Weves) [reply backend/danswer/connectors/file/connector.py:93]: It makes no assumption on how they are set!

File: backend/supervisord.conf (hunk 11 above)
Comment (Weves) [reply backend/supervisord.conf]: Sure, will just roll both. Don't think we need to keep everything around

**Pull request merged**